ios/tools/setup_tool/setup_cm.sh

```
@@ -0,0 +1,242 @@
+
+# Helper function to add items to an array in plist
+add_to_plist_array() {
+    local key=$1
+    local items=("${@:2}")
+    local current_array
+    
+    current_array=$(safe_extract_plist "$key" json)
+    if [ $? -ne 0 ] || [ -z "$current_array" ]; then
+        plutil -replace "$key" -json "$(printf '"%s",' "${items[@]}" | sed 's/,$//' | awk '{print "["$0"]"}')" "$info_plist"
+    else
+        local new_array=$(echo "$current_array" | sed 's/\[//;s/\]//;s/"//g' | tr ',' '\n' | sort -u)
```

Shellcheck

[warning] Declare and assign separately to avoid masking return values.

(SC2155)